### PR TITLE
batches: use templating for the default branch name

### PR DESCRIPTION
**Pull request by LawnGnome**
OK, so this one is for debate, not necessarily something to be immediately merged.

Every batch change needs a unique name. Every branch needs a unique name. I don't know about y'all, but I pretty much always use the batch change name as the branch name in the changeset template.

This PR changes the default template we use when creating a batch change to use the batch change name as the Git branch as well. As an added bonus, this introduces a new user to the concept of template variables.

The downside is obviously complexity: now there's another thing the user might need to understand.

Thoughts?

cc: @malomarrec @danielmarquespt 

## Test plan

I mean, I tried it. 🤷 

## App preview:

- [Web](https://sg-web-aharvey-default-branch-name.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.01 kb) | 0.00% (+0.01 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/aharvey/default-branch-name/compare-report.html;tab=live_object) for a full comparison between the commits 4d81b6f and 357efa4 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by eseliger — APPROVED** [review]
feels sane to me :)

File: client/web/src/enterprise/batches/batch-spec/edit/library/hello-world.batch.yaml [hunk 1]
```diff
@@ -15,6 +15,6 @@ steps:
 changesetTemplate:
   title: Hello World
   body: My first batch change!
-  branch: hello-world # Push the commit to this branch.
+  branch: ${{ batch_change.name }} # Push the commit to this branch.
```
[the remark this answers is not in the document]
Comment (danielmarquespt) [reply client/web/src/enterprise/batches/batch-spec/edit/library/hello-world.batch.yaml:18]: Other than this, seems ok to me :)

[the remark this answers is not in the document]
Comment (adeola-ak) [reply client/web/src/enterprise/batches/batch-spec/edit/library/hello-world.batch.yaml:18]: maybe `# Optional: Push the commit to a branch named after this batch change by default.`

**Comment by adeola-ak** [comment]
I like the idea! and i dont think it's too complex where it has a real downside

**Review by adeola-ak — APPROVED** [review]

**Review by Piszmog — APPROVED** [review]

File: client/web/src/enterprise/batches/batch-spec/edit/library/hello-world.batch.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (malomarrec) [reply client/web/src/enterprise/batches/batch-spec/edit/library/hello-world.batch.yaml:18]: :+1:

**Review by BolajiOlajide — APPROVED** [review]

File: client/web/src/enterprise/batches/batch-spec/edit/library/hello-world.batch.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LawnGnome) [reply client/web/src/enterprise/batches/batch-spec/edit/library/hello-world.batch.yaml]: > maybe # Optional: Push the commit to a branch named after this batch change by default.


I like that. Updated!

**Pull request merged**